pino plugin logs extra undefined values

**Describe the bug**
The pino plugin is logging extra `undefined` values specifically for version 4 of pino.

For example, adding the following line to packages/datadog-plugin-pino/test/index.spec.js causes the test cases to fail only for pino version 4:
```javascript
expect(record).to.have.deep.property('msg', 'message')
```
Sample output:
```
  1) Plugin
       pino
         with pino 4 (4.0.0)
           with configuration
             should add the trace identifiers to logger instances:

      AssertionError: expected { Object (pid, hostname, ...) } to have deep property 'msg' of 'message', but got 'message undefined undefined undefined undefined undefined undefined undefined undefined undefined undefined'
      + expected - actual

      -message undefined undefined undefined undefined undefined undefined undefined undefined undefined undefined
      +message

  2) Plugin
       pino
         with pino 4 (4.17.6)
           with configuration
             should add the trace identifiers to logger instances:

      AssertionError: expected { Object (level, time, ...) } to have deep property 'msg' of 'message', but got 'message undefined undefined undefined undefined undefined undefined undefined undefined undefined undefined'
      + expected - actual

      -message undefined undefined undefined undefined undefined undefined undefined undefined undefined undefined
      +message
```

--- a/packages/datadog-plugin-pino/src/index.js
+++ b/packages/datadog-plugin-pino/src/index.js
@@ -18,7 +18,7 @@ function createWrapGenLog (tracer, config) {
       const log = genLog(z)
 
       return function logWithTrace (a, b, c, d, e, f, g, h, i, j, k) {
-        const args = [a, b, c, d, e, f, g, h, i, j, k]
+        const args = [a, b, c, d, e, f, g, h, i, j, k].slice(0, arguments.length)
 
         if (!a) {
           args[0] = {}

--- a/packages/datadog-plugin-pino/test/index.spec.js
+++ b/packages/datadog-plugin-pino/test/index.spec.js
@@ -50,6 +50,7 @@ describe('Plugin', () => {
             const record = JSON.parse(stream.write.firstCall.args[0].toString())
 
             expect(record).to.not.have.property('dd')
+            expect(record).to.have.deep.property('msg', 'message')
           })
         })
       })
@@ -72,6 +73,8 @@ describe('Plugin', () => {
               trace_id: span.context().toTraceId(),
               span_id: span.context().toSpanId()
             })
+
+            expect(record).to.have.deep.property('msg', 'message')
           })
         })
       })

